rust/drivers/datafusion/src/lib.rs

```diff
@@ -707,39 +789,81 @@ pub struct DataFusionStatement {
     ctx: Arc<SessionContext>,
     sql_query: Option<String>,
     substrait_plan: Option<Plan>,
+    bound_record_batch: RefCell<Option<RecordBatch>>,
```

Comment:
I don't think you need a `RefCell` here. Because you are moving the `RecordBatch` (in in `bind` and out in `execute_update`), you can just use `Option`.
```suggestion
    bound_record_batch: Option<RecordBatch>,
```

Reply:
cool, didn't know that was possible with Options. fixed.